addons/mail/controllers/discuss.py

```diff
@@ -414,6 +414,9 @@ def channel_ping(self, channel_id, rtc_session_id=None, check_rtc_session_ids=No
     def mail_thread_data(self, thread_model, thread_id, request_list, **kwargs):
         res = {}
         thread = request.env[thread_model].with_context(active_test=False).search([('id', '=', thread_id)])
+        if not thread:
```

Comment:
We recently fixed another issue in the access rules to avoid the user to write on a private task whom doesn't belong to himself because the user cannot see the private tasks of others users.

Btw, I don't think the problem is the one you mentioned in your commit.
I think the problem is: *when the user creates a private task and assign another user to that task and then removes himself and save the task, the current user should no longer access to that task since the task becomes this private task of the other user assigned on it. So when the current user changes anything in that task, it will have a Access Error*. Here, you also have a traceback because I guess the user has no longer access to chatter and so the controller not find any thread.

If @seb-odoo is agreed with this fix to avoid having any traceback then okay for me but I will just ask you to reword the commit message like this:
```txt
[FIX] mail: check if the user can access to the thread

Before this commit, when the user creates a private task 
and assign another user to that task and then removes 
himself and save the task, the current user should no 
longer access to that task since the task becomes this 
private task of the other user assigned on it. 
So when the current user changes anything in that task, 
it will have a `Access Error`. 
However if the user tries to remove a follower on that 
task he will also have a traceback because no thread is found. 
The reason why no thread is found is because the user has 
no longer access to the record and so he has no longer 
access to the chatter.

This commit checks if a thread is found before continuing 
the process. If no thread is found then the user has no 
access to the chatter instead of throwing a traceback 
because no thread is found.

task-2857229

Co-authored-by: Prakash Prajapati <ppr@odoo.com>
```

Reply:
@xavierbol , Done!!